lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResultImpl.cs

```diff
@@ -43,20 +45,57 @@ internal class HttpClientHelper : IDriverHttpEndpoint
         private const string AppJson = "application/json";
 
         private string _driverUrl;
+        private string _appId;
 
         private readonly HttpClient _client;
+        private readonly IREEFClient _reefClient;
 
-        internal HttpClientHelper()
+        internal JobSubmissionResultImpl(IREEFClient reefClient, string filePath, bool yarn)
         {
+            _reefClient = reefClient;
             _client = new HttpClient
             {
                 Timeout = TimeSpan.FromSeconds(SecondsForHttpClientTimeout),
             };
             _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(AppJson));
+
+            if (yarn)
+            {
+                _driverUrl = GetDriverUrlForYarn(filePath);
+            }
+            else
+            {
+                _driverUrl = GetDriverUrlForLocalRuntime(filePath);
+            }
+        }
+
+        /// <summary>
+        /// Returns http end point of the web server running in the driver
+        /// </summary>
+        public string DriverUrl { get { return _driverUrl; }
         }
 
-        public string DriverUrl { get { return _driverUrl; } }
+        /// <summary>
+        /// Get application Id returned from Yarn job submission
+        /// </summary>
+        public string AppId
+        {
+            get { return _appId; }
+        }
 
+        /// <summary>
+        /// Get application final status from Yarn
```

Comment:
So, this only works on YARN? How do we communicate this to the user of this API?